federation/pkg/kubefed/init/init.go

```diff
@@ -173,6 +176,7 @@ func (o *initFederationOptions) Bind(flags *pflag.FlagSet, defaultServerImage, d
 	flags.Int32Var(&o.apiServerNodePortPort, apiserverPortFlag, 0, "Preferred port to use for api server nodeport service (0 for random port assignment). Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false, "Enables HTTP Basic authentication for the federation-apiserver. Defaults to false.")
 	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false, "Enables token authentication for the federation-apiserver. Defaults to false.")
+	flags.StringVar(&o.tolerationsString, "tolerations", "", "If specified, the pod's tolerations. Example \"key1=value1:NoSchedule,key2=value2:NoSchedule\"")
```

Comment:
@tengqm Just as PR #50815 said,

> The `kubefed init` can not specify `tolerations`, this will cause that I cannot deploy my federation panel on node which has `taint`, it is better to enable the `kubefed init` can specify the `tolerations` so that I can deploy federation panel on node which have `taint` that the federation panel can tolerate.

> this is about the federation api server?
> The intent is to have federation-apiserver deployment tolerate some taints?

Also with controller manager. Yes, add some taints tolerations.

> Why don't we just encode some tolerations into the API server's deployment instead of exposing these as user visible options?

I vote for providing a way for customization. Just encoding into the deployment will not work as expected as we don't know the specific taints, such as the key name, value and TaintEffect.

@gyliu513 Correct me if I'm wrong. PTAL.